private/bufpkg/buflock/buflock.go

```diff
@@ -142,3 +151,15 @@ func ExternalConfigDependencyV1Beta1ForDependency(dep Dependency) ExternalConfig
 type ExternalConfigVersion struct {
 	Version string `json:"version,omitempty" yaml:"version,omitempty"`
 }
+
+// isBufDugest returns true when the digest string appears to be a (older) buf
+// digest. These digests look like "b[0-9]-".
+func isBufDigest(digest string) bool {
+	if len(digest) >= 3 {
+		prefix := digest[0:3]
+		if prefix == "b1-" || prefix == "b2-" || prefix == "b3-" {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
I was hoping to keep the two packages distinct, but it's not a bad choice even if it's a bit wasteful (repeated digest parsing).

Reply:
To avoid repeated parsing you can do:

```diff
- if !isBufDigest(digest) {
-   digest = ""
- }
- // use digest
- // parse digest
+ parsedDigest, err := manifest.NewDigestFromString(digest)
+ if err != nil {
+   digest = ""
+ }
+ // use digest or parsedDigest
```